File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        vector<int> result;
        vector<vector<int>> bucket(nums.size()+1);
        unordered_map <int, int> umap;

        for (auto &i: nums) umap[i]+=1;

        for (auto &i: umap) bucket[i.second].push_back(i.first);

        for (int i = nums.size(); i >= 0; --i) {
            for (int num : bucket[i]) {
                result.push_back(num);
                k--;
                if (k == 0) return result;
            }
        }

        return {};
    }
};

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        track = {}
        result = []
        
        for n in nums:
            if track.get(n):
                track[n] += 1
            else:
                track[n] = 1

        top_k = sorted(track.items(), key=lambda x: x[1], reverse=True)
        for i in range(k):
            result.append(top_k[i][0])

        return result